modules/test/playwright/tests/export-import-service/main/remoteStaging.spec.ts

```diff
@@ -45,6 +48,325 @@ export const test = mergeTests(
+				});
+
+				apiHelpers.data.push({id: site.id, type: 'site'});
+
+				remoteSite = await remoteApiHelpers.headlessSite.createSite({
+					name: site.name,
+				});
+
+				remoteApiHelpers.data.push({id: remoteSite.id, type: 'site'});
+
+				remoteUrl = remoteApiHelpers.baseUrl.substring(
+					0,
+					remoteApiHelpers.baseUrl.length - 3
+				);
+
+				featureFlagEnabled =
+					await remoteApiHelpers.featureFlag.isFeatureFlagEnabled(
+						'LPD-35914'
+					);
+
+				if (!featureFlagEnabled.featureFlag.enabled) {
+					await remoteApiHelpers.featureFlag.updateFeatureFlag(
+						'LPD-35914',
+						true,
+						true
+					);
+				}
+
+				await apiHelpers.jsonWebServicesStaging.enableRemoteStaging({
+					groupId: site.id,
+					remoteGroupId: remoteSite.id,
+					remotePort,
+				});
+
+				for (const i of [1, 2, 3]) {
+					let layout =
+						await apiHelpers.jsonWebServicesLayout.addLayout({
+							groupId: site.id,
+							title: `Page ${i}`,
+						});
+
+					layouts.push(layout);
+
+					for (const j of [1, 2]) {
+						layout =
+							await apiHelpers.jsonWebServicesLayout.addLayout({
+								groupId: site.id,
+								parentLayoutId: layout.layoutId,
+								title: `Page ${i}${j}`,
+							});
+
+						layouts.push(layout);
+
+						if (i === 1 && j === 1) {
+							layout =
+								await apiHelpers.jsonWebServicesLayout.addLayout(
+									{
+										groupId: site.id,
+										parentLayoutId: layout.layoutId,
+										title: 'Page 111',
+									}
+								);
+							layouts.push(layout);
+						}
+					}
+				}
+			});
+
+			const pageNumbers = [1, 11, 111, 12, 2, 21, 22, 3, 31, 32];
+			await test.step('Setup pages with web content display', async () => {
+				for (const layout of layouts) {
+					await pageEditorPage.goto(layout, site.friendlyUrlPath);
+
+					await widgetPagePage.addPortlet('Web Content Display');
+					await widgetPagePage.addPortlet('Web Content Display');
+				}
+			});
+			const webContentTitle = getRandomString();
+			await test.step('Create structures, templates and web contents', async () => {
+				const fields: Array<any> = [];
+
+				for (const num of pageNumbers) {
+					fields.push({name: `Openpage${num}`, repeatable: false});
+					fields.push({name: `URL${num}`, repeatable: false});
+				}
+
+				const structureName = getRandomString();
+				const dataDefinition = getDataStructureDefinition({
+					defaultLanguageId: 'en_US',
+					fields,
+					name: structureName,
+				});
+				const structure = await apiHelpers.dataEngine.createStructure(
+					site.id,
+					dataDefinition
+				);
+
+				let i = 0;
+				const contentFields: Array<any> = [];
+				for (const layout of layouts) {
+					contentFields.push({
+						name: `Openpage${pageNumbers[i]}`,
+						value: layout.nameCurrentValue,
+					});
+					contentFields.push({
+						name: `URL${pageNumbers[i]}`,
+						value:
+							`/web${site.friendlyUrlPath}` + layout.friendlyURL,
+					});
+					i++;
+				}
```

Comment:
This could be simplified using a [flatmap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)
```
	const contentFields = layouts.flatMap((layout, i) => [
					{
						name: `Openpage${pageNumbers[i]}`,
						value: layout.nameCurrentValue,
					},
					{
						name: `URL${pageNumbers[i]}`,
						value: `/web${site.friendlyUrlPath}${layout.friendlyURL}`,
					},
				]);
```

What do you think?